build.gradle

```diff
@@ -0,0 +1,292 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+buildscript {
+	repositories {
+		maven { url "https://repo.spring.io/plugins-release" }
+	}
+	dependencies {
+		classpath("io.spring.gradle:propdeps-plugin:0.0.10.RELEASE")
+		classpath("io.spring.gradle:dependency-management-plugin:1.0.8.RELEASE")
+		classpath("org.asciidoctor:asciidoctor-gradle-plugin:1.6.1")
+		classpath("org.jfrog.buildinfo:build-info-extractor-gradle:4.9.9")
+		classpath("io.spring.nohttp:nohttp-gradle:0.0.3.RELEASE")
+	}
+}
+
+// NoHttp has to be applied at the root level
+// so that it reads all the root files, including the gradle ones.
+apply plugin: "io.spring.nohttp"
+
+checkstyle {
+	toolVersion = 8.16
+}
+
+configure(allprojects) {
+	group = "io.pivotal.spring.cloud"
+
+	apply plugin: "java-library"
+	apply plugin: "maven"
+	apply plugin: "eclipse"
+	apply plugin: "idea"
+	apply plugin: "jacoco"
+	apply plugin: "checkstyle"
+	apply plugin: "propdeps"
+	apply plugin: "propdeps-idea"
+	apply plugin: "propdeps-eclipse"
+	apply plugin: "io.spring.dependency-management"
+
+	ext {
+		springCloudVersion = "Greenwich.SR3"
+		springBootVersion = project.findProperty("springBootVersion") ?: "2.1.8.RELEASE"
+		springVersion = project.findProperty("springVersion") ?: "5.1.9.RELEASE"
+		cloudfoundryCertificateTrusterVersion = "1.0.1.RELEASE"
+		jerseyClientVersion = "1.19.4"
+		javaCfenvVersion = "1.1.2.BUILD-SNAPSHOT"
+	}
+
+	repositories {
+		mavenCentral()
+		maven { url "https://repo.spring.io/libs-release" }
+	}
+
+	dependencies {
+		testRuntimeOnly("io.spring.nohttp:nohttp:0.0.3.RELEASE")
+	}
+
+	if (project.hasProperty("springVersion") || project.hasProperty("springBootVersion") ) {
+		repositories {
+			maven { url "https://repo.spring.io/libs-snapshot" }
+		}
+	}
+
+	ext.javadocLinks = [
+		"https://docs.oracle.com/javase/8/docs/api/",
+		"https://docs.spring.io/spring/docs/${springVersion}/javadoc-api/",
+	] as String[]
+}
+
+subprojects {
+	apply from: "${rootProject.projectDir}/publish-maven.gradle"
+}
+
+configure(subprojects - [project(":spring-cloud-services-starter-config-client"),
+						 project(":spring-cloud-services-starter-service-registry"),
+						 project(":spring-cloud-services-dependencies")]) {
+	compileJava {
+		sourceCompatibility=1.8
+		targetCompatibility=1.8
+		options.encoding = "UTF-8"
+	}
+	compileTestJava {
+		sourceCompatibility=1.8
+		targetCompatibility=1.8
+		options.encoding = "UTF-8"
+	}
+
+	[compileJava, compileTestJava]*.options*.compilerArgs = [
+		"-Xlint:serial",
+		"-Xlint:varargs",
+		"-Xlint:cast",
+		"-Xlint:classfile",
+		"-Xlint:dep-ann",
+		"-Xlint:divzero",
+		"-Xlint:empty",
+		"-Xlint:finally",
+		"-Xlint:overrides",
+		"-Xlint:path",
+		"-Xlint:-processing",
+		"-Xlint:static",
+		"-Xlint:try",
+		"-Xlint:fallthrough",
+		"-Xlint:rawtypes",
+		"-Xlint:deprecation",
+		"-Xlint:unchecked",
+		"-Xlint:-options",
+	]
```

Comment:
How did we decide on this list by the way? Why not `-Xlint:all`?

Reply:
Some projects have to disable `-Xlint:-processing` because you get warnings about the annotation processor. I don't know why `options` is disabled here. Turn them on and see if you get warnings?